src/lib/services/index.ts

```diff
@@ -149,11 +153,15 @@ export const createServices = (
     const healthService = new HealthService(stores, config);
     const userFeedbackService = new UserFeedbackService(stores, config);
     const segmentService = new SegmentService(stores, config); // TODO coupled with enterprise feature
+    const changeRequestAccessReadModel = db
+        ? createChangeRequestAccessReadModel(db, config)
+        : createFakeChangeRequestAccessService();
```

Comment:
This explains why the 2 constructor methods, but still I'm not convinced we should have this logic in this file. Despite this, I see we're already doing that with `exportImportService`. At first sight, this seems to be a blurry boundary between test code and production code and even if it's now limited to this file, this may leak in the future and cause issues. In any case, I think this is a different topic and we can talk about it separately from this PR :+1:

Reply:
This logic doesn't belong here. There's a comment I made a few lines below. It's a transitional pattern before we migrate to fully feature driven architecture. After the migration we won't need this shared services file anymore. It was only useful in the technology driven architecture. And at the feature dependency graph level there will always be separate graphs for SQL based deps and in-memory based deps.

Reply:
It's fine if it's a transitional pattern, but what puzzles me is the fact that we can use this code with or without a DB. It's like implying: when there's no db we're in test, which means we're mixing production code with test code

Reply:
With the technology driven architecture we were injecting either real stores or fake stores to the big services factory. So you could also pass stores with DB an without DB. But now with the migration to feature driven architecture this is the best I found so far before we get rid of a big services factory.

Reply:
I'm ok with this as a temporal solution, but I wouldn't like this pattern to spread. The main difference with https://github.com/Unleash/unleash/blob/main/src/test/fixtures/store.ts is that it's in a `src/test` folder which should not be accessible from production code (but from a quick test I did that doesn't seem to be the case... :disappointed:)

Reply:
So if it's visible anyway it's better to have it in one place at least :)

Reply:
Maybe the difference is that now in test code you need to import the appropriate code: https://github.com/Unleash/unleash/blob/9cdd870365035077359fc98afbd5659777784aa7/src/lib/metrics.test.ts#L12 which then is used to create the objects in test code: https://github.com/Unleash/unleash/blob/9cdd870365035077359fc98afbd5659777784aa7/src/lib/metrics.test.ts#L28-L30

In the new solution, the dependency is magically changed/injected depending on the db. I'd rather have the db as non-optional and keep doing what we were doing, but I do not consider that a blocker for this PR